Better workflow

Chocolate4U opened the pull request:
## Description

Current workflow relies on deprecated Node 12 actions which has been out of support since [April 2022](https://github.com/nodejs/Release/#end-of-life-releases). This PR will:
* Update these actions to versions that run on Node 16
* Update Go version to 1.19

## Additional Information

[GitHub Actions: All Actions will begin running on Node16 instead of Node12](https://github.blog/changelog/2022-09-22-github-actions-all-actions-will-begin-running-on-node16-instead-of-node12/)

Chocolate4U committed:
Update constants.py

Chocolate4U committed:
Update main.py

Chocolate4U committed:
Update constants.py

Chocolate4U committed:
Update create_config.py

Chocolate4U committed:
Update main.yml

Chocolate4U committed:
Update create_config.py

Chocolate4U committed:
Merge pull request #1 from chocolate4u/clash-refactor

Clash refactor

Chocolate4U committed:
Update README.md

Chocolate4U committed:
Merge pull request #2 from chocolate4u/clash-refactor

Update README.md

Chocolate4U committed:
Update README.fa.md

Chocolate4U committed:
Merge pull request #3 from chocolate4u/clash-refactor

Update README.fa.md

Chocolate4U committed:
Update main.yml

Chocolate4U committed:
Update main.yml

Chocolate4U committed:
Revert main.yml

Chocolate4U committed:
Merge pull request #4 from chocolate4u/clash-refactor

Clash refactor

Chocolate4U committed:
Merge branch 'bootmortis:main' into main

Chocolate4U committed:
Update main.yml

actions/checkout@v2 > actions/checkout@v3
actions/setup-go@v3 > actions/setup-go@v4
actions/setup-python@v2 >actions/setup-python@v4
use latest versions of Python3.11 and Go

Chocolate4U committed:
Update requirements.txt

update packages

Chocolate4U committed:
Create go.sum

copy go.sum from bootmortis/sing-geosite

bootmortis commented on .github/workflows/main.yml:
```diff
@@ -16,17 +16,17 @@ jobs:
+        uses: actions/setup-go@v4
         with:
-          go-version: 1.18
+          go-version: 'stable'
```
```suggestion
          go-version: 1.19
```

bootmortis reviewed (changes requested):
I'm [not ok with changing the Python version](https://www.youtube.com/watch?v=DgAa08nSyHo).

But about Go, let's stick with `1.19`, ([the version that v2fly uses](https://github.com/v2fly/domain-list-community/blob/8cb0ee0a9ed4587c37cdbee2f937a304ed080a4f/go.mod#L3))

And about caching, we are using two Go-based projects, sing-box, and v2fly, so let's just waste GithHub runners for now.

So let's only stick with these:
- actions/checkout@v3
- actions/setup-python@v4
- actions/setup-go@v4
- go-version: 1.19

And revert other changes.

Chocolate4U committed:
Update main.yml

actions/checkout@v2 > actions/checkout@v3
actions/setup-go@v3 > actions/setup-go@v4
actions/setup-python@v2 >actions/setup-python@v4
disabled caching of go modules

Chocolate4U committed:
Revert changes to requirements.txt

Chocolate4U committed:
Delete go.sum

Chocolate4U committed:
Revert requirements.txt
